samples/hello-oboe/src/main/cpp/PlayAudioEngine.cpp

```diff
@@ -178,27 +187,18 @@ PlayAudioEngine::onAudioReady(oboe::AudioStream *audioStream, void *audioData, i
     Trace::beginSection("numFrames %d, Underruns %d, buffer size %d",
                         numFrames, underrunCountResult.value(), bufferSize);
 
+    bool is16BitFormat = (audioStream->getFormat() == oboe::AudioFormat::I16);
```

Comment:
If the stream is 16-bit then render into the buffer, then use `convertFloatToPCM16` on it